Meta/CMake/unicode_data.cmake

```
@@ -1,139 +1,124 @@
 
 set(CLDR_NUMBERS_SOURCE cldr-numbers-modern)
-set(CLDR_NUMBERS_PATH ${LOCALE_DATA_CACHE_LOCATION}/${CLDR_NUMBERS_SOURCE})
+set(CLDR_NUMBERS_PATH ${CLDR_PATH}/${CLDR_NUMBERS_SOURCE})
 
-if (ENABLE_UNICODE_DATABASE_DOWNLOAD)
-    if (NOT EXISTS ${UNICODE_DATA_PATH})
-        message(STATUS "Downloading UCD UnicodeData.txt from ${UNICODE_DATA_URL}...")
-        file(DOWNLOAD ${UNICODE_DATA_URL} ${UNICODE_DATA_PATH} INACTIVITY_TIMEOUT 10)
-    endif()
-    if (NOT EXISTS ${SPECIAL_CASING_PATH})
-        message(STATUS "Downloading UCD SpecialCasing.txt from ${SPECIAL_CASING_URL}...")
-        file(DOWNLOAD ${SPECIAL_CASING_URL} ${SPECIAL_CASING_PATH} INACTIVITY_TIMEOUT 10)
-    endif()
-    if (NOT EXISTS ${DERIVED_GENERAL_CATEGORY_PATH})
-        message(STATUS "Downloading UCD DerivedGeneralCategory.txt from ${DERIVED_GENERAL_CATEGORY_URL}...")
-        file(DOWNLOAD ${DERIVED_GENERAL_CATEGORY_URL} ${DERIVED_GENERAL_CATEGORY_PATH} INACTIVITY_TIMEOUT 10)
-    endif()
-    if (NOT EXISTS ${PROP_LIST_PATH})
-        message(STATUS "Downloading UCD PropList.txt from ${PROP_LIST_URL}...")
-        file(DOWNLOAD ${PROP_LIST_URL} ${PROP_LIST_PATH} INACTIVITY_TIMEOUT 10)
-    endif()
-    if (NOT EXISTS ${DERIVED_CORE_PROP_PATH})
-        message(STATUS "Downloading UCD DerivedCoreProperties.txt from ${DERIVED_CORE_PROP_URL}...")
-        file(DOWNLOAD ${DERIVED_CORE_PROP_URL} ${DERIVED_CORE_PROP_PATH} INACTIVITY_TIMEOUT 10)
-    endif()
-    if (NOT EXISTS ${DERIVED_BINARY_PROP_PATH})
-        message(STATUS "Downloading UCD DerivedBinaryProperties.txt from ${DERIVED_BINARY_PROP_URL}...")
-        file(DOWNLOAD ${DERIVED_BINARY_PROP_URL} ${DERIVED_BINARY_PROP_PATH} INACTIVITY_TIMEOUT 10)
-    endif()
-    if (NOT EXISTS ${PROP_ALIAS_PATH})
-        message(STATUS "Downloading UCD PropertyAliases.txt from ${PROP_ALIAS_URL}...")
-        file(DOWNLOAD ${PROP_ALIAS_URL} ${PROP_ALIAS_PATH} INACTIVITY_TIMEOUT 10)
-    endif()
-    if (NOT EXISTS ${PROP_VALUE_ALIAS_PATH})
-        message(STATUS "Downloading UCD PropertyValueAliases.txt from ${PROP_VALUE_ALIAS_URL}...")
-        file(DOWNLOAD ${PROP_VALUE_ALIAS_URL} ${PROP_VALUE_ALIAS_PATH} INACTIVITY_TIMEOUT 10)
-    endif()
-    if (NOT EXISTS ${SCRIPTS_PATH})
-        message(STATUS "Downloading UCD Scripts.txt from ${SCRIPTS_URL}...")
-        file(DOWNLOAD ${SCRIPTS_URL} ${SCRIPTS_PATH} INACTIVITY_TIMEOUT 10)
+function(remove_unicode_data_if_version_changed version version_file cache_path)
+    set(version_differs YES)
+
+    if (EXISTS ${version_file})
+        file(STRINGS ${version_file} active_version)
+        if (version STREQUAL active_version)
+            set(version_differs NO)
+        endif()
     endif()
-    if (NOT EXISTS ${SCRIPT_EXTENSIONS_PATH})
-        message(STATUS "Downloading UCD ScriptExtensions.txt from ${SCRIPT_EXTENSIONS_URL}...")
-        file(DOWNLOAD ${SCRIPT_EXTENSIONS_URL} ${SCRIPT_EXTENSIONS_PATH} INACTIVITY_TIMEOUT 10)
+
+    if (${version_differs})
```

Prefer to just name the variable alone if the [documentation](https://cmake.org/cmake/help/latest/command/if.html) indicates `<variable|string>`:

```suggestion
    if (version_differs)
```

An unquoted variable substitution (aka list expansion) into an `if()` command *can* do some unintended things.